rustls-mbedcrypto-provider/src/sign.rs

```diff
@@ -125,47 +138,47 @@ const RSA_SIGNATURE_SCHEME_PREFER_LIST: &[SignatureScheme] = &[
 
 impl rustls::sign::SigningKey for MbedTlsPkSigningKey {
     fn choose_scheme(&self, offered: &[SignatureScheme]) -> Option<Box<dyn rustls::sign::Signer>> {
-        match self.pk_type {
-            mbedtls::pk::Type::Rsa | mbedtls::pk::Type::RsaAlt | mbedtls::pk::Type::RsassaPss => {
-                // choose a rsa schema
-                for scheme in RSA_SIGNATURE_SCHEME_PREFER_LIST {
-                    if offered.contains(scheme) {
-                        let signer = MbedTlsSigner(Arc::clone(&self.pk), *scheme);
-                        return Some(Box::new(signer));
-                    }
-                }
-                None
-            }
-            mbedtls::pk::Type::Eckey | mbedtls::pk::Type::EckeyDh | mbedtls::pk::Type::Ecdsa => {
-                let scheme = self
-                    .ec_signature_scheme
-                    .expect("validated");
-                if offered.contains(&scheme) {
-                    let signer = MbedTlsSigner(Arc::clone(&self.pk), scheme);
-                    return Some(Box::new(signer));
-                }
-                None
-            }
-            _ => None,
-        }
+        let scheme = get_signature_schema_from_offered(
+            self.pk_type,
+            offered,
+            self.ec_signature_scheme,
+            self.rsa_scheme_prefer_order_list,
+        )?;
+        let signer = MbedTlsSigner(Arc::clone(&self.pk), scheme);
+        Some(Box::new(signer))
     }
 
     fn algorithm(&self) -> rustls::SignatureAlgorithm {
         self.signature_algorithm
     }
 }
 
-fn pk_type_to_signature_algo(pk_type: mbedtls::pk::Type) -> rustls::SignatureAlgorithm {
-    use rustls::SignatureAlgorithm;
+/// Helper function to choose proper [`SignatureScheme`] based on given inputs
+pub fn get_signature_schema_from_offered(
```

Comment:
Do we want this to be public as well?

Reply:
Yes, or maybe it's better to put it in `utils` crate?

Reply:
I don't have a preference in that case. I think here is fine.

Reply:
Could you write a doc comment specifying what the function does. (e.g., when it returns `None`, when it expects `ec_signature_scheme` to be `Some(..)`, etc.). Also could you explain the arguments as well?

Reply:
Just added. I just moved it to utils crate